<a href="https://colab.research.google.com/github/TedFess/DSPS_tFessaras/blob/master/HW3/lab1_Z_Test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import pandas as pd
import numpy as np
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [0]:
precipitation = pd.read_csv("https://raw.githubusercontent.com/fedhere/DSPS/master/labs/1865331.csv")

In [0]:
precipitation.describe()

PRCP        SNOW        SNWD        WESD        WESF
count  3538.000000  2266.00000  216.000000  175.000000  256.000000
mean      0.137340     0.09594    1.026852    0.160571    0.073828
std       0.382174     0.64966    2.738447    0.457994    0.175507
min       0.000000     0.00000    0.000000    0.000000    0.000000
25%       0.000000     0.00000    0.000000    0.000000    0.000000
50%       0.000000     0.00000    0.000000    0.000000    0.000000
75%       0.070000     0.00000    0.350000    0.100000    0.100000
max       6.960000    12.30000   16.400000    2.400000    1.400000

In [0]:
precipitation.shape

(3538, 8)

In [0]:
rainonly = precipitation[["DATE", "PRCP"]]

In [0]:
rainonly["date"] = pd.to_datetime(rainonly.DATE)

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [0]:
recentonly = rainonly[rainonly['date'] > pd.to_datetime("2018/01/01")].describe()

In [0]:
recentonly

PRCP
count  615.000000
mean     0.162504
std      0.365007
min      0.000000
25%      0.000000
50%      0.000000
75%      0.120000
max      2.450000

In [0]:
z_test_result = (0.162504 - 0.137340) / (0.382174/np.sqrt(615))

In [0]:
print(z_test_result)

1.6328868686034312


**Conclusion:** Based upon the results of my Z-test result of 1.633 with a confidence  *p = 0.05* we cannot reject the null hypothesis, therefore I find that the time period 2018/01/01 to today does not have enough evidence to say there is a statistically significant different amount of rain.